projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/utils.ts

```diff
@@ -12,20 +13,25 @@ import {
  * @return The sections for the My Jetpack tab panel.
  */
 export function getMyJetpackSections(): TabPanelProps[ 'tabs' ] {
-	return [
+	const { userIsAdmin } = getMyJetpackWindowInitialState();
+
+	const tabs = [
 		{
 			name: MY_JETPACK_SECTION_OVERVIEW,
 			title: __( 'Overview', 'jetpack-my-jetpack' ),
 		},
 		{
 			name: MY_JETPACK_SECTION_PRODUCTS,
 			title: __( 'Products', 'jetpack-my-jetpack' ),
+			isAdminOnly: true,
 		},
 		{
 			name: MY_JETPACK_SECTION_HELP,
 			title: __( 'Help', 'jetpack-my-jetpack' ),
 		},
 	];
+
+	return tabs.filter( tab => ! ( tab.isAdminOnly && ! userIsAdmin ) );
 }
```

Comment:
Instead of adding that flag to the object in the array here, we could define this in `constants.ts`

```ts
export const ADMIN_ONLY_SECTIONS = [ MY_JETPACK_SECTION_PRODUCTS ];
```

Then use it here

```ts
import { currentUserCan } from '@automattic/jetpack-script-data';
import { ADMIN_ONLY_SECTIONS } from './constants';

const tabs = [
	// Tabs here
];

if ( currentUserCan( 'manage_options' ) ) {
	return tabs;
}

return tabs.filter( tab => ! ADMIN_ONLY_SECTIONS.includes( tab.name ) );
```

Reply:
It would have been better if we hadn't lost the types for that tabs array, which was previously ensured by the return type.